src/System.Linq/src/System/Linq/Intersect.cs

```diff
@@ -19,8 +19,14 @@ public static IEnumerable<TSource> Intersect<TSource>(this IEnumerable<TSource>
             {
                 throw Error.ArgumentNull(nameof(second));
             }
-
-            return IntersectIterator(first, second, null);
+           
+            foreach (TSource element in first)
```

Comment:
@JonHanna  This was the error I said before. Already fixed via Distinct(). I am not sure what you mean in your previous comment.